hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -255,11 +257,13 @@ private[hudi] object HoodieSparkSqlWriter {
     }
 
     if (!tableExists) {
-      val archiveLogFolder = parameters.getOrElse(
-        HoodieTableConfig.HOODIE_ARCHIVELOG_FOLDER_PROP_NAME, "archived")
+      val archiveLogFolder = parameters.get(HoodieTableConfig.HOODIE_ARCHIVELOG_FOLDER_PROP_NAME).get
```

Comment:
can we back out the changes that are not strictly needed for this PR. (same comment on block above)

Reply:
this is because in HoodieIndexUtils.getIndexType will also get the INDEX_TYPE_PROP , and should add "parameters.getOrElse(
       HoodieIndexConfig.INDEX_TYPE_PROP, HoodieIndexConfig.DEFAULT_INDEX_TYPE)" here.  as your comment ,can move the two param init to HoodieWriterUtils. parametersWithWriteDefaults